lib/hotel_manager.rb

```diff
@@ -0,0 +1,132 @@
+require_relative "hotel_block"
+
+module Hotel
+  class HotelManager
+    attr_reader :rooms, :blocks, :total_reservations
+
+    def initialize
+      @rooms = []
+      @blocks = []
+      @total_reservations = 0
+    end
+
+    def initialize_rooms(num)
+      if num.class != Integer
+        raise ArgumentError.new("Integer value must be provided. Value provided: '#{num}'")
+      else
+        num.times do |i|
+          room = Hotel::Room.new((i + 1), 200)
+          @rooms << room
+        end
+      end
+    end
+
+    def list_all_rooms
+      return @rooms
+    end
+
+    def reserve_room(date_range, room = nil)
+      picked_room = ""
+      if !room
+        available_rooms = list_available_rooms(date_range)
+        raise ArgumentError.new("No rooms available") if available_rooms == []
+        picked_room = available_rooms[0]
+      else
+        picked_room = find_room(room)
+      end
+
+      @total_reservations += 1
+      reservation = Reservation.new(date_range, @total_reservations, picked_room.number, picked_room.cost)
+
+      find_room(reservation.room_number).reservations << reservation
```

Comment:
A very small design thing to consider in the future:

Since your `HotelManager` class manages number of total reservations, it makes sense that `@total_reservations += 1` logic lives here. But since `Room` manages reservations, too, it might be interesting to see the above lines take more of an approach to:

1. Find the right room in `HotelManager`, THEN
2. Tell the right room to create and add a reservation.

I would personally achieve this by doing the `find_room` method first, and then creating an instance method in `Room` to create the `Reservation` object instance and shovel it into its `reservations` array. This is a thought for a future refactoring!